packages/property-view/src/browser/resource-property-view/resource-property-data-service.spec.ts

```diff
@@ -0,0 +1,127 @@
+    ...ApplicationProps.DEFAULT.frontend.config
+});
+
+import { Container } from '@theia/core/shared/inversify';
+import { ResourcePropertyDataService } from './resource-property-data-service';
+import { FileService } from '@theia/filesystem/lib/browser/file-service';
+import URI from '@theia/core/lib/common/uri';
+import { PropertyDataService } from '../property-data-service';
+import { FileSelection } from '@theia/filesystem/lib/browser/file-selection';
+import { Navigatable } from '@theia/core/lib/browser/navigatable';
+import { FileStat } from '@theia/filesystem/lib/common/files';
+
+disableJSDOM();
+
+let resourcePropertyDataService: ResourcePropertyDataService;
+
+const mockFileStat: FileStat = {
+    isFile: false,
+    isDirectory: true,
+    isSymbolicLink: false,
+    resource: new URI('resource'),
+    name: 'name'
+};
+
+describe('resource-property-data-service', () => {
+
+    before(() => {
+        disableJSDOM = enableJSDOM();
+        const container = new Container();
+        container.bind(ResourcePropertyDataService).toSelf().inSingletonScope();
+        container.bind(FileService).toConstantValue({
+            async resolve(uri: URI): Promise<FileStat> {
+                return mockFileStat;
+            }
+        } as FileService);
+        container.bind(PropertyDataService).to(ResourcePropertyDataService).inSingletonScope();
+        resourcePropertyDataService = container.get(ResourcePropertyDataService);
+    });
+
+    after(() => {
+        disableJSDOM();
+    });
+
+    const fileSelection: FileSelection = {
+        fileStat: mockFileStat
+    };
+
+    const navigatableSelection: Navigatable = {
+        getResourceUri(): URI | undefined {
+            return new URI('resource-uri');
+        },
+        createMoveToUri(): URI | undefined {
+            return new URI('move-uri');
+        }
+    };
+
+    const arrayFileSelection: FileSelection[] = [fileSelection, { fileStat: { ...mockFileStat, resource: new URI('secondURI') } }];
+
+    describe('#canHandle', () => {
+
+        it('should not handle an empty object selection', () => {
+            expect(resourcePropertyDataService.canHandleSelection({})).eq(0);
+        });
+
+        it('should not handle an undefined selection', () => {
+            expect(resourcePropertyDataService.canHandleSelection(undefined)).eq(0);
+        });
+
+        it('should handle a file selection', () => {
+            expect(resourcePropertyDataService.canHandleSelection(arrayFileSelection)).to.be.greaterThan(0);
+        });
+
+        it('should handle a navigatable selection', () => {
+            expect(resourcePropertyDataService.canHandleSelection(navigatableSelection)).to.be.greaterThan(0);
+        });
+    });
+
+    describe('#providePropertyData', () => {
+
+        it('should return the file-stat of a file selection', async () => {
+            const data = await resourcePropertyDataService.providePropertyData(arrayFileSelection);
```

Comment:
This test is identical to line 109, I believe we mean to test the single item array and not both `arrayFileSelection`?

Reply:
Yes, we meant that, I just changed it thank you !